python/lsst/daf/persistence/swiftStorage.py

```diff
@@ -0,0 +1,519 @@
+        each location in butlerLocation.getLocations()
+        """
+        localFile = self.getLocalFile(butlerLocation.getLocations()[0])
+        butlerLocation.locationList = [localFile.name]
+        butlerLocation.storage = PosixStorage('/')
+        obj = butlerLocation.storage.read(butlerLocation)
+        return obj
+
+    def _getLocalFile(self, location):
+        """Implementation of getLocalFile that does not wrap the swift
+        ClientException so that this function may be used by this class in
+        various functions that want to handle the exception directly.
+
+        Parameters
+        ----------
+        location : string
+            A location of the the file in storage, relative to the storage's
+            root.
+
+        Raises
+        ------
+        swift.ClientException
+            Indicates an error downloading the object.
+        """
+        location = self.getFitsHeaderStrippedPath(location)[0]
+        localFile = self.fileCache.get(location, None)
+        if not localFile:
+            localFile = tempfile.NamedTemporaryFile(
+                suffix=os.path.splitext(location)[1])
+            headers, contents = self._connection.get_object(self._containerName,
+                                                            location)
+            localFile.write(contents)
+            localFile.flush()
+            self.fileCache[location] = localFile
+        localFile.seek(0)
+        return localFile
+
+    def getLocalFile(self, location):
+        """TODO this is changed to return a handle to a local temproary file.
+        If we keep it there's a few places that will have to be modified.
+
+        As it is expected the local file will be read (not written to), the
+        current position in the file is set to 0 before it is returned.
+
+        The local file will be deleted when the file object is deleted, so this
+        function does not close the file unless an error is raised internally.
+        Callers may close or explicitly delete the object when they are done
+        with it or may allow it to be garbage collected.
+
+        Parameters
+        ----------
+        location : string
+            A location of the the file in storage, relative to the storage's
+            root.
+
+        Returns
+        -------
+        A handle to a local copy of the file. If storage is remote it will be
+        a temporary file. If storage is local it may be the original file or
+        a temporary file. The file name can be gotten via the 'name' property
+        of the returned object.
+
+        Raises
+        ------
+        RuntimeError
+            If there is an error downloading the object.
+        """
+        self._log.info("SwiftStorage getting file {}".format(location))
+        start = time.time()
+        try:
+            f = self._getLocalFile(location)
+            self._log.info(
+                "...getting file took {} seconds.".format(time.time() - start))
```

Comment:
Would debug be better, or what would you suggest?
For the time being I find it very useful for profiling Butler behavior, although I agree it will end up being noisy. I'd like to leave something that will print during unit test execution, but I don't need it to be 'info'.